includes/collection/class-inbox.php

```diff
@@ -83,6 +83,17 @@ public static function add( $activity, $recipients ) {
 		$title      = self::get_object_title( $activity->get_object() );
 		$visibility = is_activity_public( $activity ) ? ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC : ACTIVITYPUB_CONTENT_VISIBILITY_PRIVATE;
 
+		/*
+		 * For activities with an 'instrument' property (e.g., QuoteRequest), we store
+		 * the instrument URL as the object_id. This allows efficient querying by instrument.
+		 * For all other activities, we store the object URL as before.
+		 */
+		if ( $activity->has( 'instrument' ) && $activity->get_instrument() ) {
+			$object_id = object_to_uri( $activity->get_instrument() );
+		} else {
+			$object_id = object_to_uri( $activity->get_object() );
+		}
```

Comment:
Is this a situation where `$object_id = object_to_uri( $activity->get_instrument() ?: $activity->get_object() );` could work?